### PR TITLE
refactor(admin): refactor admin consumers/plugin_metadata

### LINKED ISSUE #8569: feat: As a user, I want to refactor the resources under apisix.admin, so that we don't need to repeat
### Description

Currently, all the resources under `apisix.admin` are basically copied and pasted from a template and changed. If you make general changes to the resource, such as fix count or add adapter, you need to change it file by file. And due to historical problems, some resource implementation methods are incomplete.
I hope we can refactor the code to something like `core.config.new`, where we only need to pass a few parameters to create a new resource.

Note: Please read every resource file before jumping into the refactor. Some resources have particular logic. We need a smart way to handle them.

CC @An-DJ

### PR DESCRIPTION
**Pull request by An-DJ**
### Description

Fixes parts of #8569 
- [X] consumers
- [X] plugin_metadata

### Checklist

- [X] I have explained the need for this PR and the problem it solves
- [X] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [X] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/admin/resource.lua [hunk 1]
```diff
@@ -31,27 +31,32 @@ local mt = {
 }
 
 
+local no_id_res = {"consumers", "plugin_metadata"}
```
Comment (spacewander) [inline apisix/admin/resource.lua:34]: We can define it as hash to speed up the lookup?

File: apisix/admin/resource.lua [hunk 2]
```diff
@@ -31,27 +31,32 @@ local mt = {
...
-    end
+        if need_id and conf.id and tostring(conf.id) ~= tostring(id) then
+            return nil, {error_msg = "wrong ".. self.kind .. " id"}
+        end
 
-    conf.id = id
+        conf.id = id
+    end
 
     core.log.info("schema: ", core.json.delay_encode(self.schema))
```
Comment (spacewander) [inline apisix/admin/resource.lua:61]: Look like in these resources, there is some redundant code? For example, the scheme will be logged twice.

File: apisix/admin/consumers.lua [hunk 3]
```diff
@@ -16,25 +16,13 @@
...
 
-local _M = {
-    version = 0.1,
-    need_v3_filter = true,
-}
 
-
-local function check_conf(username, conf)
-    -- core.log.error(core.json.encode(conf))
-    if not conf then
```
[the remark this answers is not in the document]
Comment (An-DJ) [reply apisix/admin/consumers.lua:31]: It seems that the other resources may check additional conditions or parameters before checking the schema.

Like the routes:
https://github.com/apache/apisix/blob/master/apisix/admin/routes.lua#L36

For these reasons, schema check cannot be moved to public code for the time being.

[the remark this answers is not in the document]
Comment (An-DJ) [reply apisix/admin/consumers.lua:31]: The same goes for secrets:

https://github.com/apache/apisix/blob/master/apisix/admin/secrets.lua#L58

Its schema is decided by `typ`.

[the remark this answers is not in the document]
Comment (spacewander) [reply apisix/admin/consumers.lua:31]: There is a common check in the https://github.com/apache/apisix/blob/bc37568b0975edb7f111038610b66f5a4d908230/apisix/admin/resource.lua#L36

We don't need to repeat it in the custom checker. It's not about schema.

**Review by spacewander — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**